### DISCUSSION TITLE
Change all Javascript (admin/storefront) files to typescript with corresponding types

### DISCUSSION THREAD

**Discussion by AndreasA** [shopware/platform#3152] [RFC / Proposal]
**Effort**: not sure.

### Description: 
If all files use types and typescript it is easy for developers to also use typescript but still possible to keep using plain javascript. However, the opposite just is not possible as the necessary types are missing.

Or at least add corresponding types which might be necessary anyway for globals like Shopware.Component in administration.

From what I can see this has already been started in the adminsitration but a lot of files are still ".js" files and also even if there are ".ts" files do not mean that there are corresponding types or that types have been exported. Furthermore, it adds better auto completion support for e.g. Shopware.Component.override etc.

### Benefits
See above.

### Break strategy
shouldn't really break anything as component override/extend and service decorations should stay the same. same for storefront plugin extension.

**Comment by tobiasberge** [comment]
Hi @AndreasA,

like you pointed out we already started introducing TypeScript in the administration and we want to extend this further in the future to have even more type checking / better auto completion.

We can also imagine this for the Storefront e.g. for core the Plugin, PluginManager classes, helpers, utilities etc. Normally this shouldn't be a break for the handling of the modules/functions in general. The only break I could imagine is importing core modules explicitly with `.js` extension. We have to evaluate if it's possible without break and if webpack is able to load `.ts` even when `.js` is imported.

**Comment by lasomethingsomething** [comment]
Thank you for your proposal, @AndreasA. :) Due to lack of activity around this in recent years and the likelihood that this won't change anytime soon, we'll close this proposal for now.
